Src/Web/Web.Shared.Net/ApplicationInsightsHttpModule.cs

```diff
@@ -92,13 +93,41 @@ private void OnBeginRequest(object sender, EventArgs eventArgs)
 
                 this.TraceCallback("OnBegin", httpApplication);
 
+                this.HookOnSendRequestHeaders(httpApplication);
+
                 if (this.requestModule != null)
                 {
                     this.requestModule.OnBeginRequest(httpApplication.Context);
                 }
 
                 // Kept for backcompat. Should be removed in 2.3 SDK
-                WebEventsPublisher.Log.OnBegin();    
+                WebEventsPublisher.Log.OnBegin();
+            }
+        }
+
+        /// <summary>
+        /// When sending the response headers, allow request module to add the IKey's target hash.
+        /// </summary>
+        /// <param name="httpApplication">HttpApplication instance.</param>
+        private void HookOnSendRequestHeaders(HttpApplication httpApplication)
+        {
+            // We use reflection here because 'AddOnSendingHeaders' is only available post .net framework 4.5.2. Hence we call it if we can find it.
+            MethodInfo addOnSendingHeadersMethod = httpApplication.Response.GetType().GetMethod("AddOnSendingHeaders");
```

Comment:
have try{}catch to handle exceptions and keep working even if this failed. Also check `httpApplication` for `null`, same as `Response`